src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1298 @@
+            prevalence_of_wasting = self.wasting_models.get_wasting_prevalence(agegp=agegp).predict(df.loc[mask])
+
+            # categorize into moderate (-3<=WHZ<-2) or severe (WHZ<-3) wasting
+            wasted = self.rng.random_sample(len(prevalence_of_wasting)) < prevalence_of_wasting
+            for idx in prevalence_of_wasting.index[wasted]:
+                probability_of_severe = self.get_odds_probs_wasting(agegp=agegp)
+                wasted_category = self.rng.choice(['WHZ<-3', '-3<=WHZ<-2'], p=[probability_of_severe,
+                                                                               1 - probability_of_severe])
+                df.at[idx, 'un_WHZ_category'] = wasted_category
+                df.at[idx, 'un_last_wasting_date_of_onset'] = self.sim.date
+                df.at[idx, 'un_ever_wasted'] = True
+                # start without treatment
+                df.at[idx, 'un_am_treatment_type'] = 'none'
+
+        # ------------------------------------------------------------------
+        # # # # # # Give MUAC category, presence of oedema, and determine
+        # acute malnutrition state # # # # #
+        self.population_poll_clinical_am(df)
+
+        children_with_sam = df.loc[df.is_alive & (df.age_exact_years < 5)
+                                   & (df.un_clinical_acute_malnutrition == 'SAM')]
+        for person in children_with_sam.index:
+            # update clinical symptoms for severe wasting
+            self.wasting_clinical_symptoms(person_id=person)
 
     def initialise_simulation(self, sim):
+        """Prepares for simulation:
+        * Schedules the main polling event
+        * Schedules the main logging event
+        """
+
+        # schedule wasting pool event
+        sim.schedule_event(WastingPollingEvent(self), sim.date + DateOffset(months=3))
+
+        # schedule wasting logging event
+        sim.schedule_event(WastingLoggingEvent(self), sim.date + DateOffset(months=12))
+
+    def on_birth(self, mother_id, child_id):
+        """Initialise properties for a newborn individual.
+        :param mother_id: the mother for this child
+        :param child_id: the new child
+        """
+
+        df = self.sim.population.props
+
+        # Set initial properties
+        df.at[child_id, 'un_ever_wasted'] = False
+        df.at[child_id, 'un_WHZ_category'] = 'WHZ>=-2'  # not undernourished
+        df.at[child_id, 'un_clinical_acute_malnutrition'] = 'well'
+        df.at[child_id, 'un_last_wasting_date_of_onset'] = pd.NaT
+        df.at[child_id, 'un_acute_malnutrition_tx_start_date'] = pd.NaT
+        df.at[child_id, 'un_sam_death_date'] = pd.NaT
+        df.at[child_id, 'un_am_bilateral_oedema'] = False
+        df.at[child_id, 'un_am_MUAC_category'] = '>=125mm'
+        df.at[child_id, 'un_am_treatment_type'] = 'not_applicable'
+
+    def get_odds_probs_wasting(self, agegp: str, lm_scaling: bool = False) -> Union[float, int]:
```

Comment:
It would be better to call this fnc `get_probs_or_odds_wasting` since it only calculates one of those, probs (default) or odds.

Why is the input param called `lm_scaling`? Does it really scale a linear model? I would suggest using a name that indicates that the function will calculate odds instead of probs, for example, 'odds_bool'.